common/src/View/MapViewBase.h

```diff
@@ -102,6 +102,17 @@ namespace TrenchBroom {
             MapViewBase(Logger* logger, std::weak_ptr<MapDocument> document, MapViewToolBox& toolBox, Renderer::MapRenderer& renderer, GLContextManager& contextManager);
 
             void setCompass(std::unique_ptr<Renderer::Compass> compass);
+
+            /**
+             * Perform tasks that are needed for a fully initalized MapViewBase.
+             * Subclasses should call this at the end of their constructors.
+             * (Does virtual function calls, so we can't call it in the MapViewBase constructor.)
+             * 
+             * On normal app startup, these tasks are handled by documentDidChange(),
+             * but when changing map view layotus (e.g. 1 pane to 2 pane) there are 
+             * no document notifications to handle these tasks, so it must be done by the constructor.
+             */
+            void postInit();
```

Comment:
Shouldn't `documentDidChange` call this method as well? And if so, it should have a different name.

Reply:
A better name might be `mapViewBaseVirtualIInit`. It needs to communicate that it's finishing things that can't be put in the MapViewBase constructor.

We can factor out
```
createActions();
updateActionStates();
updatePickResult();
```
and share them between mapViewBaseVirtualIInit and documentDidChange into a different function.